setup/src/Magento/Setup/Model/Installer.php

```diff
@@ -842,17 +858,29 @@ private function handleDBSchemaData($setup, $type)
         $moduleNames = $this->moduleList->getNames();
         $moduleContextList = $this->generateListOfModuleContext($resource, $verType);
         foreach ($moduleNames as $moduleName) {
+            $moduleContext = $moduleContextList[$moduleName];
             $this->log->log("Module '{$moduleName}':");
             $configVer = $this->moduleList->getOne($moduleName)['setup_version'];
-            $currentVersion = $moduleContextList[$moduleName]->getVersion();
+            $currentVersion = $moduleContext->getVersion();
+
+            $event = new BeforeModuleIsProcessed($this, $setup, $moduleContext, $type);
+            $this->eventManager->trigger($event);
+
+            if ($event->propagationIsStopped()) {
```

Comment:
remove this if statement. The event should notify the system that something happened, but it should not terminate the entire process. Also, this does not work for asynchronous events. 
If you would like to terminate the process, we should introduce another extension point for that, but not events.